src/components/SettingsDialog.vue

```diff
@@ -221,25 +223,19 @@ export default {
             isOpen: false,
             printImage: false,
             recipeFolder: "",
-            resetPrintImage: true,
             showTagCloudInRecipeList: true,
-            resetTagCloud: true,
             scanningLibrary: false,
-            // By setting the reset value initially to true, it will skip one watch event
-            // (the one when config is loaded at page load)
-            resetInterval: true,
             updateInterval: 0,
             visibleInfoBlocks: [...INFO_BLOCK_KEYS],
-            resetVisibleInfoBlocks: true,
+            writeChanges: true,
         }
     },
     watch: {
         async printImage(newVal, oldVal) {
-            // Avoid infinite loop on page load and when reseting value after failed submit
-            if (this.resetPrintImage) {
-                this.resetPrintImage = false
+            if (!this.writeChanges) {
```

Comment:
We should dispatch the updated option to the store either here, or when settings dialog is closed (would be okay for the print option since its effect is not immediately visible on the screen.